app/controller/grid/Grid.js

```diff
@@ -0,0 +1,328 @@
+        // https://www.sencha.com/forum/showthread.php?301458-Loading-mask-not-setting-for-bound-store-loaded-in-ViewController.init
+        var me = this;
+
+        var view = me.getView();
+        view.setLoading('Loading Records...');
+
+        var filters = Ext.clone(store.getFilters().items);
+
+        if (me.spatialFilter) {
+            filters.push(me.spatialFilter);
+        }
+
+        var wfsGetFeatureFilter = GeoExt.util.OGCFilter.getOgcWfsFilterFromExtJsFilter(filters, 'And', '2.0.0');
+
+        if (wfsGetFeatureFilter) {
+            params.filter = wfsGetFeatureFilter;
+        }
+    },
+
+    /**
+    * Hide the loading mask when the store has loaded
+    *
+    * @private
+    */
+    onWfsStoreAfterLoad: function () {
+        var view = this.getView();
+        view.setLoading(false);
+    },
+
+    /**
+     * Store the spatial filter as a property of this class
+     * then force a reload of the grid store with the new filter
+     * Finally apply all filters to any associated layers.
+     *
+     * @param {Ext.util.Filter} spatialFilter
+     * @private
+     */
+    onSpatialFilter: function (spatialFilter) {
+
+        var me = this;
+        me.spatialFilter = spatialFilter;
+
+        // force a reload of the grid store
+        var grid = me.getView();
+        var store = grid.getStore();
+        store.loadWfs();
+
+
+        this.filterAssociatedLayers();
+    },
+    /**
+    * Enable and disable paging for the grid.
+    * Disabling paging allows all records to be loaded into the
+    * grid for an Excel export. Enabling paging improves load
+    * performance.
+    *
+    * @private
+    */
+    togglePaging: function (checkBox, checked) {
+
+        var me = this;
+        var grid = me.getView();
+
+        var pagingToolbar = grid.down('gx_wfspaging_toolbar');
+        pagingToolbar.setDisabled(!checked);
+
+        var store = grid.getStore();
+
+        // save the initial store parameters
+
+        if (!me.originalPageSize) {
+            me.originalPageSize = store.pageSize;
+        }
+
+        if (checked) {
+            store.pageSize = me.originalPageSize;
+            store.startIndex = 0; // reset each time // me.startIndex;
+        } else {
+            store.pageSize = null;
+            store.startIndex = 0;
+        }
+
+        store.loadWfs();
+    },
+
+    /**
+    * Export the current records in the grid to Excel
+    *
+    * @private
+    */
+    exportToExcel: function () {
+
+        var me = this;
+        var grid = me.getView();
+
+        if (!grid.saveDocumentAs) {
+            Ext.Msg.alert('Not Supported',
+                'The Excel export is not supported in this version of the system', Ext.emptyFn);
+            return;
+        }
+
+        grid.setLoading('Exporting to Excel...');
+
+        // later in an event listeners
+        grid.saveDocumentAs({
+            type: 'xlsx',
+            title: 'Map Export',
+            fileName: 'myExport.xlsx'
```

Comment:
Would be good to have these properties as config in the grid view class

Reply:
Added